src/pages/Chat/ChatRoom/ChatRoom.tsx

```diff
@@ -61,91 +74,97 @@ function ChatRoom({
   };
 
   const onSendHandler = () => {
-    if (text.trim()) {
-      sendMessage({ chatRoomId: roomId, text });
+    if (text.trim() && roomId && +roomId) {
+      sendMessage({ chatRoomId: +roomId, text });
     }
     setText("");
   };
 
   const defineName = (userID: number): string => {
-    if (roomUsers) {
-      return userID === roomUsers.client.id
-        ? `${roomUsers.client.firstName || "User"} ${
-            roomUsers.client.lastName || "Client"
+    if (room && room.client && room.freelancer) {
+      return userID === room.client.id
+        ? `${room.client.firstName || "User"} ${
+            room.client.lastName || "Client"
           }`
-        : `${roomUsers.freelancer.firstName || "User"} ${
-            roomUsers.freelancer.lastName || "Freelancer"
+        : `${room.freelancer.firstName || "User"} ${
+            room.freelancer.lastName || "Freelancer"
           }`;
     }
     return "User";
   };
 
   const disableInput = (): boolean => {
     if (
-      userRole === "client" &&
-      roomStatus === ChatRoomStatusEnum.CLIENT_ONLY
+      userRole === UserRoleEnum.CLIENT &&
+      room?.status === ChatRoomStatusEnum.CLIENT_ONLY
     ) {
       return false;
     }
     if (
-      userRole === "freelancer" &&
-      roomStatus === ChatRoomStatusEnum.FREELANCER_ONLY
+      userRole === UserRoleEnum.FREELANCER &&
+      room?.status === ChatRoomStatusEnum.FREELANCER_ONLY
     ) {
       return false;
     }
-    return roomStatus !== ChatRoomStatusEnum.FOR_ALL;
+    return room?.status !== ChatRoomStatusEnum.FOR_ALL;
   };
```

Comment:
move everything from line 38 to 109 to custom hook

Reply:
done